enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Installer Telemetry Metrics
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+ > 1. How to filter to collect only real user usage and prevent OpenShift developer/Red Hat internal users from skewing metrics? Perhaps filter by version?
+ > 2. Is it possible to determine whether a user modified an ignition config or manifest asset by comparing the asset the installer would generate to the one found on disk?
+ > 3. How to authenticate using pull secret.
+ 
+## Summary
+Telemetry collects metrics from running clusters, but no metrics are obtained from the installer binary.
+This enhancement proposes that the installer would push metrics related to user workflows, installation statistics, and feature usage. The installer would be instrumented to collect Prometheus label key-value pairs, convert them to metric samples and push those metrics to an [aggregation pushgateway][weaveworks-aggregation-pushgateway].
+This enhancement focuses on the design of the metrics to be pushed to Prometheus.
+
+## Motivation
+
+Obtaining metrics from the installer would allow engineers and product managers to understand different user workflows, environments, experiences, and difficulties, as well as an indication of feature usage.  By pushing these metrics to the Telemeter, OpenShift members would be able to create a variety of queries related to these aspects.
+
+### Goals
+
+1. Design meaningful metrics to collect user experience and behavior with the installer
+
+### Non-Goals
+
+1. Collected metrics will not allow analysis of individual installer invocations and will not display detailed data such as log or error messages. We won't be able to answer "how long did an individual installation take?" but rather 95% of installs finish within, say, 40 minutes.
+2. Installer instrumentation/internal implementation will be discussed in a separate design/enhancement
+3. Pushgateway implementation will be discussed in a separate design/enhancement
+
+## Proposal
+
+### User Stories
+
+#### Telemetry Analyst
+Some examples:
+
+As a telemetry analyst I would like to create queries which analyze aggregate installation data to determine how long the installation process takes in total and broken down by different stages so that I can better understand user experience, expose any abnormalities, and compare different versions of the installer.
+
+As a telemetry analyst I would like to create queries which show aggregate counts of failure by stage of installation so that we can better understand where users are experiencing problems. (This will not provide  specific error messages/logs).
```

Comment:
In https://github.com/openshift/installer/pull/3535/files the condition "reason" keys would be extremely useful to report and shouldn't be a problem for cardinality. Being able to see the rate at which specific known problems are occurring would be immensely valuable.